harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -274,14 +280,30 @@ def _run(self) -> None:
             try:
                 if w.kind == workload.Workload.Kind.RUN_STEP:
                     action = "training"
+                    metrics = self._train_for_step(
+                        w.step_id,
+                        w.num_batches,
+                        w.total_batches_processed,
+                    )
                     response = {
-                        "metrics": self._train_for_step(
-                            w.step_id,
-                            w.num_batches,
-                            w.total_batches_processed,
-                        ),
+                        "metrics": metrics,
                         "stop_requested": self.context.get_stop_requested(),
                     }  # type: workload.Response
+                    metrics = self.context.distributed.broadcast(metrics)
+                    for callback in self.callbacks.values():
+                        callback.on_training_workload_end(
+                            avg_metrics=metrics["avg_metrics"],
+                            batch_metrics=metrics["batch_metrics"],
+                        )
+                        if (
+                            self.context.distributed.size > 1
+                            and not self.context._average_training_metrics
+                        ):
+                            logging.warning(
+                                "Only the chief worker's training metrics are being reported, due "
+                                "to setting average_training_metrics to False."
+                            )
```

Comment:
Let's not log this:
- more than once per python process, or
- if the user never defined an on_training_workload_end

Reply:
Ah, this one was actually meant for [4646](https://github.com/determined-ai/determined/pull/4646), just to bring extra attention to the consequence of having `average_training_metrics` false.  Removed here and added there.  It's also intended to be shown regardless of what callbacks are used, since its importance is independent of callbacks.